tests/calc/test_calc_tools.py

```diff
@@ -1250,3 +1267,210 @@ def test_remove_nans():
+
+    assert p.shape == t.shape
+    assert_array_equal(p.metpy.x, t.metpy.x)
+    assert_array_equal(p.metpy.y, t.metpy.y)
+
+    assert m.shape == t.shape
+    assert_array_equal(m.metpy.x, t.metpy.x)
+    assert_array_equal(m.metpy.y, t.metpy.y)
+
+    assert dx.check('m')
+    assert dy.check('m')
+
+    assert_array_almost_equal(lat, data.lat, 5)
+
+
+@pytest.mark.parametrize('xy_order', (False, True))
+def test_parse_grid_arguments_cartesian(test_da_xy, xy_order):
+    """Test the operation of parse_grid_arguments with no lat/lon info."""
+    @parse_grid_arguments
+    @preprocess_and_wrap(broadcast=['scalar', 'parallel_scale', 'meridional_scale'],
+                         wrap_like=('scalar', 'dx', 'dy', 'scalar', 'scalar', 'latitude',
+                                    None, None))
+    def check_params(scalar, dx=None, dy=None, x_dim=-1, y_dim=-2,
+                     parallel_scale=None, meridional_scale=None, latitude=None):
+        return scalar, dx, dy, parallel_scale, meridional_scale, latitude, x_dim, y_dim
+
+    # Remove CRS from dataarray
+    data = test_da_xy.reset_coords('metpy_crs', drop=True)
+    del data.attrs['grid_mapping']
+
+    if xy_order:
+        data = data.transpose(..., 'x', 'y')
+
+    t, dx, dy, p, m, lat, x_dim, y_dim = check_params(data)
+    if xy_order:
+        assert x_dim == 2
+        assert y_dim == 3
+    else:
+        assert x_dim == 3
+        assert y_dim == 2
+
+    assert_array_almost_equal(t, data)
+    assert_array_almost_equal(dx, 500 * units.km)
+    assert_array_almost_equal(dy, 500 * units.km)
+
+    assert p is None
+    assert m is None
+    assert lat is None
+
+
+def test_parse_grid_arguments_unknown_dims():
+    """Test parse_grid_arguments with data with unknown dimensions."""
+    @parse_grid_arguments
+    @preprocess_and_wrap(broadcast=['scalar', 'parallel_scale', 'meridional_scale'],
+                         wrap_like=('scalar', 'dx', 'dy', 'scalar', 'scalar', 'latitude',
+                                    None, None))
+    def check_params(scalar, dx=None, dy=None, x_dim=-1, y_dim=-2, parallel_scale=None,
+                     meridional_scale=None, latitude=None):
+        return scalar, dx, dy, parallel_scale, meridional_scale, latitude, x_dim, y_dim
+
+    lat, lon = np.meshgrid(np.array([38., 40., 42]), np.array([263., 265., 267.]))
+    test_da = xr.DataArray(
+        np.linspace(300, 250, 3 * 3).reshape((3, 3)),
+        name='temperature',
+        dims=('dim_0', 'dim_1'),
+        coords={
+            'lat': xr.DataArray(lat, dims=('dim_0', 'dim_1'),
+                                attrs={'units': 'degrees_north'}),
+            'lon': xr.DataArray(lon, dims=('dim_0', 'dim_1'), attrs={'units': 'degrees_east'})
+        },
+        attrs={'units': 'K'}).to_dataset().metpy.parse_cf('temperature')
+
+    with pytest.raises(AttributeError,
+                       match='horizontal dimension coordinates cannot be found.'):
+        check_params(test_da)
```

Comment:
@jthielen @dcamron This is encoding the current behavior, which makes sense directly: the x/y coordinates are not identified. I just want to make sure we all agree that this is the correct behavior?

Reply:
Yep, I'd agree this is the intended behavior when horizontal dimension coordinates are missing!

That being said, this test is labeled "unknown dims", rather than "missing dimension coordinates". Would we be able to rename this test, then create a new one for when we have dimension coordinates, but we can't infer which is x and which is y (e.g., named `dim_0` and `dim_1`), and so have to rely on the assumed `[..., Y, X]` dimension order (i.e., testing https://github.com/Unidata/MetPy/blob/8652ec357f1e8ff0b04c5903d032dfe7e78703a2/src/metpy/calc/tools.py#L1048-L1051)?

Reply:
So currently in that case, it only works if someone manually specifies dx and dy--`grid_deltas` will error out. I *think* that's ok?